grapher/chart/ChartInterface.ts

```diff
@@ -25,4 +26,7 @@ export interface ChartInterface {
     // Todo: should all charts additionally have a placedSeries: ChartPlacedSeries[] getter?
 
     transformTable: ChartTableTransformer
+
+    yAxis?: AbstractAxis
+    xAxis?: AbstractAxis
```

Comment:
This type should maybe be `HorizontalAxis | VerticalAxis` which is more accurate and makes it clearer that they can be oriented either way.

Reply:
I'm not sure we should call them `xAxis` and `yAxis` if they can be oriented either way...

Reply:
Yeah, I was thinking of renaming to "primary" and "secondary" axis, but it doesn't seem like a straightforward decision.

To rename them, we'd have to change the Grapher configs, the Explorer configs, and make the codebase and configuration less understandable, more abstract. And all that for only one exception – `DiscreteBar` – everywhere else they match expectations.

The other option is to not allow re-orienting them – consistently using Y for vertical, X for horizontal. This would mean changing how `DiscreteBar` works and finding a different way to pass on Y axis from `LineChart` to X axis of `DiscreteBar`. (The purpose of this is to preserve the log/lin, `yAxisMin`, etc.)

From the perspective of switching charts, it feels like they should stay as they are and maybe only be renamed ("primary" and "secondary" or potentially something else). From the perspective of developer usability, e.g. if it was a library, it makes sense to have consistent X and Y.

Reply:
For backwards-compatibility I think we should keep them as they are. Migrating away from X and Y would require bigger changes.